### PR TITLE
Fix Span Debugger Display String to correctly show the string contents

### PR DESCRIPTION
**Pull request by ahsonkhan**
Resolves https://github.com/dotnet/coreclr/issues/16450

Leftover from https://github.com/dotnet/corefx/pull/26726#issuecomment-362522270

cc @stephentoub, @khellang, @pakrym, @KrzysztofCwalina

### CODE REVIEW

**Comment by pakrym** [comment]
How does it look in vs for `char` vs `byte`?

File: src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace System
     /// or native memory, or to memory allocated on the stack. It is type- and memory-safe.
     /// </summary>
     [DebuggerTypeProxy(typeof(SpanDebugView<>))]
-    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [DebuggerDisplay("{ToString(),raw}")]
```
Comment (stephentoub) [inline src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs]: By default I thought VS rendered most things with ToString().  I could be wrong, but what happens if you remove the DebuggerDisplay attribute entirely?

**Comment by ahsonkhan** [comment]
> How does it look in vs for `char` vs `byte`?

image

image

File: src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs (hunk 1 above)
Comment (pakrym) [reply src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs]: image
This happens (special characters are not escaped)

Comment (ahsonkhan) [reply src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs]: If we don't have it, we end up with the issue mentioned here: https://github.com/dotnet/corefx/pull/26726#issuecomment-362166770

image

Comment (stephentoub) [reply src/Common/src/CoreLib/System/ReadOnlySpan.Fast.cs]: Ah. Thanks.

**Comment by khellang** [comment]
What if you try with `"\r\b\0"` as in the original issue?

**Comment by ahsonkhan** [comment]
> What if you try with `"\r\b\0"` as in the original issue?

See https://github.com/dotnet/corefx/pull/27338#discussion_r169784343

**Comment by ahsonkhan** [comment]
System.Net.Requests.Tests failing on Ubuntu.1604.Amd64.Open:Release-x64 with seg fault:

https://mc.dot.net/#/user/ahsonkhan/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/fd25d3cb8fc8054d5cdc118b24e50a0f931782e0/workItem/System.Net.Requests.Tests/wilogs
```text
2018-02-21 22:08:53,266: INFO: proc(54): run_and_log_output: Output:    System.Net.Tests.FtpWebRequestTest.Ftp_AppendFile [SKIP]
2018-02-21 22:08:53,266: INFO: proc(54): run_and_log_output: Output:       Condition(s) not met: \"LocalServerAvailable\"
2018-02-21 22:08:54,630: INFO: proc(54): run_and_log_output: Output: /home/helixbot/dotnetbuild/work/96d5d89c-3b8b-4afb-9beb-f43b108fdc9f/Work/01a7818b-79e2-43b9-9067-47e346b87957/Unzip/RunTests.sh: line 88: 69406 Segmentation fault      (core dumped) $RUNTIME_PATH/dotnet xunit.console.netcore.exe System.Net.Requests.Tests.dll -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonlinuxtests -notrait category=OuterLoop -notrait category=failing
2018-02-21 22:08:54,675: INFO: proc(54): run_and_log_output: Output: Dumpling has not been installed yet. Please run "DumplingHelper.py install_dumpling" before collect_dumps.
2018-02-21 22:08:54,679: INFO: proc(54): run_and_log_output: Output: ~/dotnetbuild/work/96d5d89c-3b8b-4afb-9beb-f43b108fdc9f/Work/01a7818b-79e2-43b9-9067-47e346b87957/Unzip
2018-02-21 22:08:54,680: INFO: proc(54): run_and_log_output: Output: ----- end 22:08:54 ----- exit code 139 ----------------------------------------------------------
2018-02-21 22:08:54,682: INFO: proc(54): run_and_log_output: Output: Unable to find executable corerun
2018-02-21 22:08:54,744: INFO: proc(58): run_and_log_output: Exit Code: 139
2018-02-21 22:08:54,745: ERROR: scriptrunner(91): _main: Error: No exception thrown, but XUnit results not created
2018-02-21 22:08:54,745: ERROR: helix_test_execution(83): report_error: Error running xunit None
```

cc @davidsh, @Priya91, @wfurt

**Comment by ahsonkhan** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by ahsonkhan** [comment]
The Tizen build timed out: 
https://ci.dot.net/job/dotnet_corefx/job/master/job/tizen_armel_cross_debug_prtest/12381/

```text
Build timed out (after 120 minutes). Marking the build as aborted.
15:26:13 
Build was aborted
```

cc @dotnet/dnceng, any ideas why?

@dotnet-bot test Tizen armel Debug Build

**Pull request merged**